### PR TITLE
change flashatten to ms.nn

### PR DESCRIPTION
**Pull request by CaitinZhao**

### CODE REVIEW

**Comment by SamitHuang** [comment]
requirement needs change for it? MS2.1 ?

**Comment by CaitinZhao** [comment]
> requirement needs change for it? MS2.1 ?

it's used to compatible with the MindSpore version

File: examples/stable_diffusion_xl/gm/modules/attention.py [hunk 1]
```diff
@@ -118,7 +100,7 @@ def __init__(
 
         self.to_out = nn.SequentialCell(nn.Dense(inner_dim, query_dim), nn.Dropout(p=dropout))
 
-        self.flash_attention = FlashAttention()
+        self.flash_attention = FlashAttention(dim_head)
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/modules/attention.py:103]: stable_diffusion_xl/gm/modules/diffusionmodules/model.py 调用位置一起修改下

File: examples/stable_diffusion_xl/gm/modules/attention.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 from mindspore import nn, ops
 
 try:
-    from mindspore.ops._op_impl._custom_op.flash_attention.flash_attention_impl import get_flash_attention
+    from mindspore.nn.layer.flash_attention import FlashAttention
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/modules/attention.py:11]: 是否可以使用 nn.FlashAttention 接口

Comment (CaitinZhao) [reply examples/stable_diffusion_xl/gm/modules/attention.py:11]: 这个目前没有对外，只能用全路径import

File: examples/stable_diffusion_xl/gm/modules/attention.py (hunk 1 above)
Comment (CaitinZhao) [reply examples/stable_diffusion_xl/gm/modules/attention.py:103]: ok

**Comment by SamitHuang** [comment]
TODO: use ` nn.FlashAttention` API for future MS versions

**Pull request closed without merging**